File: php/countVowelsAndConsonants.php
<?php
  /*
  Count the vowels and consonants in a string. We'll use the Merriam-Webster
  Dictionary definition of a consonant as "a letter representing a consonant,
  usually used in English of any letter except a, e, i, o, and u".

  Created on March 4, 2020
  @author: Charles Newman (https://github.com/26point2Newms)
  */

  define("VOWELS", array('a','e','i','o','u'));
  define("DIGITS", array('0','1','2','3','4','5','6','7','8','9'));

  $srcString = "InterviEw Questions";

  // Remove spaces and convert to lower case
  $srcString = str_replace(" ", "", strtolower($srcString));
  
  $vowelCount = countVowels($srcString);
  $consonantCount = countConsonants($srcString);

  printf("\n\t%d vowels and %d consonants!\n", $vowelCount, $consonantCount);

  function countVowels($str)
  {

    $vowelCount = 0;

    $strArray = str_split($str);

    foreach ($strArray as $char)
    {
      if (in_array($char, VOWELS))
        $vowelCount++;
    }
    
    return $vowelCount;
  } 

  function countConsonants($str)
  {
    $consonantCount = 0;

    $strArray = str_split($str);

    foreach ($strArray as $char)
    {
      if (!in_array($char, VOWELS) && !in_array($char, DIGITS))
        $consonantCount++;
    }
  
    return $consonantCount;
  }
?>

File: python/countVowelsAndConsonants.py
'''
Count the vowels and consonants in a string.

Created on Jan 13, 2017

@author: Charles Newman (https://github.com/26point2Newms)
'''

VOWELS = "aeiou"
DIGITS = "0123456789"

def countVowels(srcStr):
    vowelCount = 0
    
    for c in srcStr:
        if c in VOWELS:
            vowelCount += 1
            
    return vowelCount

def countConsonants(srcStr):
    consonantCount = 0
    
    for c in srcStr:
        if not c in VOWELS and not c in DIGITS:
            consonantCount += 1
            
    return consonantCount 
        
    
def main():
    srcString = "InterviEw Questions"
    
    # Remove spaces and convert to lower case
    srcString = srcString.replace(" ", "").lower()
    
    vowelCount = countVowels(srcString)
    consonantCount = countConsonants(srcString)
    
    print(str(vowelCount) + " vowels and " + str(consonantCount) + " consonants.")
    
if __name__ == '__main__':
    main()
    